### ISSUE TITLE
Selenium 3.12.0 Windows org.openqa.selenium.json.JsonException: Cannot coerce something that is not a number to a number: STRING

### ISSUE THREAD

**Issue by dhoer** [SeleniumHQ/selenium#5910]
## Meta -
OS:  
Windows 2016/10

Selenium Version:  
3.12.0

Browser:  
N/A

## Expected Behavior -
The choco-selenium install of 3.12.0 would behave the same as 3.11.0 or earlier.

Version 3.11.0 or earlier produces this result when starting nssm:

```
Service "Selenium Hub" installed successfully!
Set parameter "AppDirectory" for service "Selenium Hub".
Set parameter "AppParameters" for service "Selenium Hub".
Set parameter "Start" for service "Selenium Hub".
Set parameter "AppStdout" for service "Selenium Hub".
Set parameter "AppStderr" for service "Selenium Hub".
Set parameter "AppRotateFiles" for service "Selenium Hub".
Selenium Hub: START: The operation completed successfully.
```

## Actual Behavior -
Get the following error when nssm tries to start selenium as a hub service with v3.12.0:

Version 3.12.0 throws the following error:

```
Service "Selenium Hub" installed successfully!
Set parameter "AppDirectory" for service "Selenium Hub".
Set parameter "AppParameters" for service "Selenium Hub".
Set parameter "Start" for service "Selenium Hub".
Set parameter "AppStdout" for service "Selenium Hub".
Set parameter "AppStderr" for service "Selenium Hub".
Set parameter "AppRotateFiles" for service "Selenium Hub".
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Selenium Hub: Unexpected status SERVICE_PAUSED in response to START control.
 at <ScriptBlock>, C:\ProgramData\chocolatey\lib\selenium\tools\chocolateyInstall.ps1: line 92
at <ScriptBlock>, C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1: line 48
```

## Steps to reproduce -

Re-run the appveyor job for 3.12.0 branch:

https://ci.appveyor.com/project/dhoer/choco-selenium/build/3.12.0-187

Only the version was updated for the 3.12.0 branch:

https://github.com/dhoer/choco-selenium/compare/3.12.0?expand=1

**Comment by barancev** [comment]
Can this issue be reproduced without choco-selenium?

**Comment by dhoer** [comment]
If you have Vagrant, you can follow the steps below and see the issue on a Windows 10 box.

https://github.com/dhoer/choco-selenium/blob/master/TESTING.md

**Comment by barancev** [comment]
I see, but the question is: what should we do to make NSSM happy? What's expected from Selenium to make it work again?

**Comment by dhoer** [comment]
Not sure if this is an NSSM problem, or if NSSM is exposing an issue with Selenium.  Even though the choco package is 6 months old, internally I have been using this setup for over 3+ years with no issues until the 3.12 release.

**Comment by dhoer** [comment]
I actually had time to look at this. Both hub and node are throwing the following:

```
Exception in thread "main" org.openqa.grid.common.exception.GridConfigurationException: Error with the JSON of the config : Cannot coerce something that is not a number to a number: STRING
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T15:15:08.936Z'
System info: host: 'VAGRANT-10', ip: '10.0.2.15', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_172'
Driver info: driver.version: unknown
	at org.openqa.grid.internal.utils.configuration.GridNodeConfiguration.loadFromJSON(GridNodeConfiguration.java:407)
	at org.openqa.grid.internal.utils.configuration.GridNodeConfiguration.loadFromJSON(GridNodeConfiguration.java:387)
	at org.openqa.grid.internal.cli.GridNodeCliOptions.toConfiguration(GridNodeCliOptions.java:205)
	at org.openqa.grid.internal.cli.GridNodeCliOptions.toConfiguration(GridNodeCliOptions.java:34)
	at org.openqa.grid.selenium.GridLauncherV3.launch(GridLauncherV3.java:115)
	at org.openqa.grid.selenium.GridLauncherV3.main(GridLauncherV3.java:82)
Caused by: org.openqa.grid.common.exception.GridConfigurationException: Cannot coerce something that is not a number to a number: STRING
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T15:15:08.936Z'
System info: host: 'VAGRANT-10', ip: '10.0.2.15', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_172'
Driver info: driver.version: unknown
	at org.openqa.grid.internal.utils.configuration.StandaloneConfiguration.loadFromJson(StandaloneConfiguration.java:179)
	at org.openqa.grid.internal.utils.configuration.GridNodeConfiguration.loadFromJSON(GridNodeConfiguration.java:392)
	... 5 more
Caused by: org.openqa.selenium.json.JsonException: Cannot coerce something that is not a number to a number: STRING
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T15:15:08.936Z'
System info: host: 'VAGRANT-10', ip: '10.0.2.15', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_172'
Driver info: driver.version: unknown
	at org.openqa.selenium.json.NumberCoercer.lambda$apply$0(NumberCoercer.java:48)
	at org.openqa.selenium.json.JsonTypeCoercer.lambda$null$6(JsonTypeCoercer.java:142)
	at org.openqa.selenium.json.JsonTypeCoercer.coerce(JsonTypeCoercer.java:122)
	at org.openqa.selenium.json.InstanceCoercer.lambda$apply$0(InstanceCoercer.java:83)
	at org.openqa.selenium.json.JsonTypeCoercer.lambda$null$6(JsonTypeCoercer.java:142)
	at org.openqa.selenium.json.JsonTypeCoercer.coerce(JsonTypeCoercer.java:122)
	at org.openqa.selenium.json.JsonInput.read(JsonInput.java:158)
	at org.openqa.grid.common.GridConfiguredJson.toType(GridConfiguredJson.java:39)
	at org.openqa.grid.internal.utils.configuration.StandaloneConfiguration.loadFromJson(StandaloneConfiguration.java:175)
	... 6 more
```

Seems similar to https://github.com/groupon/Selenium-Grid-Extras/issues/404.

**Comment by dhoer** [comment]
The issue is that Selenium 3.12.0 expects the json configuration to have port as an integer.  Previous Selenium versions allowed port to also be a string. That is no longer the case.  

The work around is to make sure the port in both hub and node json config is an integer.

**Comment by barancev** [comment]
@shs96c Simon, what do you think if we allow automatic string to number conversion in NumberCoercer?

**Comment by dhoer** [comment]
It would also be awesome to include the name value pair having issue in message: https://github.com/SeleniumHQ/selenium/blob/8943c610a80f7ce5f09798f42f2ca37225be2e0f/java/client/src/org/openqa/selenium/json/NumberCoercer.java#L49

**Comment by shs96c** [comment]
This should be fixed by 1a0f31e2997ea5856696fa4b804483f87d850d08
